clang/test/3C/typedefbounds.c

```diff
@@ -0,0 +1,17 @@
+// RUN: 3c -alltypes -addcr %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
+// RUN: 3c --addcr --alltypes %s -- | %clang_cc1  -fcheckedc-extension -x c -
+//
+// 3c -alltypes -output-postfix=checked %s
+// 3c -alltypes %S/typedefs.checked.c -- | diff %S/typedefs.checked.c -
+// rm %S/typedefs.checked.c
+
+typedef char* string;
+//CHECK_ALL: typedef _Array_ptr<char> string;
+
+void foo(void) {
+	string x = "hello";
+	//CHECK_ALL: string x : byte_count(5) = "hello";
```

Comment:
Wow that's really interesting! The typedef doesn't have the bounds; they come outside. Is it ever possible to have a typedef that includes the bounds inside it, in your setup?